### PR TITLE
Nav menu styling fixes

### PR DESCRIPTION
**Pull request by antsgar**
**Description**

- Updated active state styling for menu subitems so they're different from items.
- Removed tabs from settings since they're no longer needed (we now have direct links navigating through each section in the navigation).
- Make map full width (approach extendable to future full width pages as well) and center Redraw / Confirm buttons.

Jira link:

https://lite-farm.atlassian.net/browse/LF-3948
https://lite-farm.atlassian.net/browse/LF-3950
https://lite-farm.atlassian.net/browse/LF-3952

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

**Review by SayakaOno — COMMENTED** [review]
Looks good overall!

A few of my thoughts:
- I would like a title on the pages that had the tab (account, people and farm). Maybe it's just me, but I feel like I'm lost.
- You are still waiting for the decision on the small modal, but I think the longer modal like below looks nicer when it is on the map...
Screenshot 2024-01-03 at 2 36 47 PM

**Review by kathyavini — COMMENTED** [review]
It looks good and I like the full-width solution!!

I checked the UX channel because I was curious about the buttons and modal going inside the map header now, but looks like no reply yet. I think specifically the modal might need an outline in this case, like the top and side menu bars have, because otherwise the contrast is disconcertingly low:

Screenshot 2024-01-03 at 2 49 25 PM

Other than that, I see a TypeError crash when drawing a line (posted in Slack), but locally I don't see it anymore after merging integration into this branch; could you confirm that from your end too? If the header styling gets the a-ok from UX and the crash is resolved with an integration merge, then this has my approval 🙂 (since I won't be here tomorrow to approve in actuality)

**Comment by antsgar** [comment]
@SayakaOno I switched back to the buttons / modal being absolutely positioned so you should now see the modal over the map. For the pages titles, I'd like Loïc's input on that so I'd prefer we defer that to a future update, would you be okay with that?

As for the error in the map, I'm able to repro, it seems this is related to the Google Map React library + React 18 + Strict Mode (see here https://github.com/google-map-react/google-map-react/issues/1129). I'm not sure how to disable strict mode with Vite, but I'm also able to repro on integration without these changes so I don't think it's something new, and if it's related to Strict Mode it should just happen locally.

**Review by Duncan-Brain — CHANGES_REQUESTED** [review]
I am seeing the error as well + "history.push is not a function" on areas and lines and everything. But I think it is because the history prop should be destructured from the arrow function -- that seems to work for me to fix all type errors. Will point it out inline. ... I think history is available through some parent component or something for the other routes but I am not 100% on that.

I am technically fine with the change to pass the isCompactSideMenu down as props in the route... but it is the only Route we do that for. So I am not a huge fan of the change of pattern. Just wondering if there is some sort of 100% width solution instead that will respond with it since it should just filling the content container. -- I am fine if thats not feasible for the release but ya seems like a new pattern to pass props this way.

File: packages/webapp/src/routes/index.jsx
Comment (Duncan-Brain) [inline packages/webapp/src/routes/index.jsx:515]: @antsgar I think history just needs to be destructured here is all to fix the errors. My only real change request

Comment (antsgar) [reply packages/webapp/src/routes/index.jsx:515]: Ahhh good catch! Will fix. Re passing isCompactSideMenu -- if the styling is done properly, we shouldn't even need to be aware of the width of the side menu. Sadly the absolute positioning of elements here in the map makes it hard to center stuff without relying on that number. Even then, instead of just drilling that down we should store it globally with React Context, but I figured this was okay for now.

Comment (antsgar) [reply packages/webapp/src/routes/index.jsx]: @Duncan-Brain should be fixed now, thank you for catching it!

**Review by Duncan-Brain — APPROVED** [review]
Works for me!

**Review by SayakaOno — APPROVED** [review]
The modal looks great!
As for the pages titles, I am okay with leaving it as a future improvement :)

Thank you for looking into the error, I will create a ticket!

**Pull request merged**
